Gems/Atom/RHI/Metal/Code/Source/RHI/BindlessArgumentBuffer.cpp

```diff
@@ -94,23 +98,40 @@ namespace AZ::Metal
         else
         {
             m_boundedArgBuffer = ArgumentBuffer::Create();
-            //For the bounded approach we have one AB hold all the bindless resource types
-            for (uint32_t i = 0; i < static_cast<uint32_t>(RHI::ShaderResourceGroupData::BindlessResourceType::Count); ++i)
+            //For the bounded approach we have one AB that holds all the bindless resource types
+            for (uint32_t i = 0; i < static_cast<uint32_t>(BindlessResourceType::Count); ++i)
             {
                 MTLArgumentDescriptor* textureArgDescriptor = [[MTLArgumentDescriptor alloc] init];
```

Comment:
Nit: textureArgDescriptor doesn't seem a good name since it can be for a texture or a buffer